### PR TITLE
Feature/zoom beyond max

### PR DESCRIPTION
**Pull request by merlos**
Removes the limitation on the tile maps camera distance. 

When the max zoom is reached, larger zoom tiles are extracted by interpolating from the max zoom tile. This is a feature that comes out of the box in MapCache 0.8.0

Fixes in some sort of way the comment https://github.com/merlos/iOS-Open-GPX-Tracker/issues/166#issuecomment-670101206

"modify the design of OSM and display the names of features a bit bigger. It’s really hard to read on iOS with Retina display."

### CODE REVIEW

**Comment by vincentneo** [comment]
I haven't played around with this yet, but just to add, another way to help with the hard to read issue could be to introduce retina tiles, but not on OSM (not supported) rather like for CARTO, as I experimented in my Avenue app thing.

**Comment by merlos** [comment]
MapCache has support for setting the tile size

File: Pods/MapCache/MapCache/Classes/CachedTileOverlay.swift [hunk 1]
```diff
@@ -10,43 +10,105 @@ import MapKit
...
+                let size = MKMapSize(width: adjustedTileSize / Double(scale),
+                                     height: adjustedTileSize / Double(scale))
+                let tileRect = MKMapRect(origin: point, size: size)
+                
+                // check that a portion of the tile intersects with the maps rect
+                // no need to do the work on a tile that won't be seen
+                guard rect.intersects(tileRect) else { continue }
+                
+                let path =  MKTileOverlayPath(x: x, y: y, z: z, contentScaleFactor: scale)
+                let tile = ZoomableTile(maximumZPath: path, rect: tileRect, overZoom: Zoom(overZoom))
```
Comment (vincentneo) [inline Pods/MapCache/MapCache/Classes/CachedTileOverlay.swift:108]: Got a crash at

Comment (vincentneo) [reply Pods/MapCache/MapCache/Classes/CachedTileOverlay.swift:108]: Only seems to happen if the tile I'm using is OpenTopoMap, zooming in as much as I could. Tested and replicable on both iOS 13.4.1 and iOS 14 GM

Comment (vincentneo) [reply Pods/MapCache/MapCache/Classes/CachedTileOverlay.swift:108]: I believe it happens as `overzoom` can reach 256 / 512, which is more than what `UInt8` `Zoom` can handle.

**Comment by vincentneo** [comment]
Looks good to me!

**Pull request merged**